### ISSUE TITLE
Block buffers are wrong sizes in NSDataAdditions initWithBase64EncodedString

### ISSUE THREAD

**Issue by akempgen** [colloquy/colloquy#778]
*Issue migrated from trac ticket [#513](http://colloquy.info/project/ticket/513)*

**type:** Defect | **component:** Chat Core (IRC) | **severity:** Blocker | **priority:** Normal | **version:** 2.0 (Mac) | **status:** closed | **resolution:** Fixed
___
#### 2006-02-25 23:35:35: chris@… created the issue

On line 26 of NSDataAdditions.m, the sizes of the inbuf and outbuf arrays are swapped. If you're decoding base 64, inbuf should have 4 elements, and outbuf should have 3. This code does not correctly decode base 64 in its current state...

basically, it reads:
unsigned char inbuf3, outbuf4;

instead, it should read:
unsigned char inbuf4, outbuf3;
